File: 98. Validate Binary Search Tree/fault.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @description https://leetcode.com/problems/validate-binary-search-tree/
 * @type self
 * @param {TreeNode} root
 * @return {boolean}
 */
var isValidBST = function(root) {
  if (!root.left && !root.right) {
    return true;
  }
  if (!root.left) {
    return root.val < root.right.val && isValidBST(root.right);
  }
  if (!root.right) {
    return root.val > root.left.val && isValidBST(root.left);
  }
  return root.val > root.left.val && root.val < root.right.val && isValidBST(root.left) && isValidBST(root.right);
};

File: 1123. Lowest Common Ancestor of Deepest Leaves/fault.java
/**
 * https://leetcode.com/problems/lowest-common-ancestor-of-deepest-leaves/
 * self
 */
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
/**
 * https://leetcode.com/problems/lowest-common-ancestor-of-deepest-leaves/
 * self
 */
class Solution {
  public TreeNode lcaDeepestLeaves(TreeNode root) {
    HashMap<TreeNode, Integer> map = new HashMap<>();
    TreeNode node = root;
    int depth = 0;
    map.put(node, depth);
    if (node.left != null) {
      map.put(node.left, depth++);
      node = node.left;
    }
    if (node.right != null) {
      map.put(node.right, depth++);
      node = node.right;
    }
    return node;
  }
}

// same with 865: https://leetcode.com/problems/smallest-subtree-with-all-the-deepest-nodes/